File: backend/app/routes/notifications.py
from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity

from app import db
from app.models import Notification

notifications_bp = Blueprint("notifications", __name__)


@notifications_bp.route("", methods=["GET"])
@jwt_required()
def list_notifications():
    user_id = int(get_jwt_identity())
    page = request.args.get("page", 1, type=int)
    per_page = request.args.get("per_page", 20, type=int)

    query = Notification.query.filter_by(user_id=user_id).order_by(
        Notification.created_at.desc()
    )
    pagination = query.paginate(page=page, per_page=per_page, error_out=False)

    return jsonify(
        {
            "notifications": [n.to_dict() for n in pagination.items],
            "total": pagination.total,
            "page": pagination.page,
            "pages": pagination.pages,
        }
    )


@notifications_bp.route("/unread-count", methods=["GET"])
@jwt_required()
def unread_count():
    user_id = int(get_jwt_identity())
    count = Notification.query.filter_by(user_id=user_id, is_read=False).count()
    return jsonify({"count": count})


@notifications_bp.route("/<int:notification_id>/read", methods=["PUT"])
@jwt_required()
def mark_read(notification_id):
    user_id = int(get_jwt_identity())
    n = db.session.get(Notification, notification_id)

    if not n or n.user_id != user_id:
        return jsonify({"errors": ["通知不存在"]}), 404

    n.is_read = True
    db.session.commit()
    return jsonify({"notification": n.to_dict()})


@notifications_bp.route("/read-all", methods=["PUT"])
@jwt_required()
def mark_all_read():
    user_id = int(get_jwt_identity())
    Notification.query.filter_by(user_id=user_id, is_read=False).update(
        {"is_read": True}
    )
    db.session.commit()
    return jsonify({"message": "全部已读"})

File: frontend/src/api/notifications.ts
import client from './client';
import type { Notification } from '../types';

interface NotificationListResponse {
  notifications: Notification[];
  total: number;
  page: number;
  pages: number;
}

export const notificationsApi = {
  list: (page = 1, perPage = 20) =>
    client.get<NotificationListResponse>('/notifications', {
      params: { page, per_page: perPage },
    }),

  unreadCount: () =>
    client.get<{ count: number }>('/notifications/unread-count'),

  markRead: (id: number) =>
    client.put<{ notification: Notification }>(`/notifications/${id}/read`),

  markAllRead: () =>
    client.put<{ message: string }>('/notifications/read-all'),
};
